vtex/faststore
# Feat: Adds `ProductGallery` Sections + `Filter` component from CMS

## What's the purpose of this pull request?

Create CMS sections for `ProductGallery` and integrate the data.
** The Filter component is part of ProductGallery section

## How it works?

- [x] Creates sections in `sections.json` for `ProductGallery` and its component's such as `Filter`.
- [x] Create `content-types` settings for `sort default value` and `items per page` in Search Page and PLP.
- [x] Integrate the `SearchInput` from `Navbar` Section with `sort` param in the result links. 
- [x] adapt pages (`PLP` and `Search`), and `SearchInput` to receive CMS props.
- [x] Refactor new `ProductGallery` component instead of compound.

After the merge
- [ ] sync in main workspace after merge this PR.
- [ ] publish changes in Search Page in main workspace.
- [ ] publish changes in PLP Page in main workspace.
- [ ] publish changes in Global Sections Page in main workspace.

## How to test it?

Unfortunately, you can't test everything altogether like clicking in the Search result link and see the Search Page, you need to check each preview by time.
You can use my dev workspace in this draft version and see the preview version:
- You can test the `PLP` using the PLP page in CMS - https://formspace--storeframework.myvtex.com/admin/new-cms/faststore/plp/edit/568e09a3-03cf-48ce-9a0c-363008ad8681/
- You can test the `Search Page` using the Search Page page in CMS -  https://formspace--storeframework.myvtex.com/admin/new-cms/faststore/search/edit/e9e9dfb0-2acf-4990-8983-0b19b63d7b26/
- You can test the `GlobalSection -> Navbar -> Input Search` Links by hovering the mouse over the `SearchHistory` and `SearchTop` and checking the sort param - https://formspace--storeframework.myvtex.com/admin/new-cms/faststore/globalSections/edit/a56db112-87a6-40d5-a4d5-f6df686f93d0/

You can also create a new workspace and test using the preview:

1. You can test it running core package locally: `yarn dev`.
2. Create a new workspace with `vtex use <name>` in `storeframework` account.
3. run `vtex cms sync faststore` to get the updated sections and content-type.
4. Add the `ProductGallery` as a draft in PLP page or Search Page, you can change the default values and the settings for each page (items per page and sort configs).
5. Click in the preview link (use the same content-type and versionId etc for localhost) and make sure all the preview cookie was set.
6. browse to a new PLP, or Search Page depending on the current preview link page and see the changes (don't need to publish).


### Starters Deploy Preview

TBD after this PR got merged.
